### PR TITLE
Add an API for Interests

### PR DESCRIPTION
**Pull request by DGoedtkindtEPFL**

### CODE REVIEW

File: app/src/main/java/com/github/polybooks/core/database/Database.kt [hunk 1]
```diff
@@ -15,6 +15,11 @@ interface Database {
      * */
     val saleDatabase : SaleDatabase
 
+    /**
+     * The instance of a Interest Database associated with this Database
+     * */
+    val interestDatabase : InterestDatabase
+
```
Comment (Snowflake8) [inline app/src/main/java/com/github/polybooks/core/database/Database.kt:22]: A voir si ça marche avec FireBase

File: app/src/main/java/com/github/polybooks/core/User.kt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package com.github.polybooks.core
+
+/**
+ * This class contains all the information to describe a user.
+ * */
+data class User(val uid : Int, val pseudo : String)
```
[the remark this answers is not in the document]
Comment (RaulinN) [reply app/src/main/java/com/github/polybooks/core/User.kt:6]: Can you think of something else? Even in the ER-model, we didn't plan for anything else

File: app/src/main/java/com/github/polybooks/core/database/InterestDatabase.kt [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    /**
+     * List all the Semesters in the database.
+     * */
+    fun listAllSemesters() : CompletableFuture<List<Semester>>
+
+    /**
+     * List all the Courses in the database.
+     * */
+    fun listAllCourses() : CompletableFuture<List<Field>>
```
Comment (RaulinN) [inline app/src/main/java/com/github/polybooks/core/database/InterestDatabase.kt:29]:
```suggestion
    /**
     * List all the Courses in the database.
     * */
    fun listAllCourses() : CompletableFuture<List<Course>>
```

**Review by RaulinN — CHANGES_REQUESTED** [review]

**Review by RaulinN — APPROVED** [review]
okay with me now :)

File: app/src/main/java/com/github/polybooks/core/User.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (Snowflake8) [reply app/src/main/java/com/github/polybooks/core/User.kt:6]: Interests no?

**Review by Snowflake8 — APPROVED** [review]
Simpler interface, i agree

**Review by Huntsekkern — APPROVED** [review]

**Comment by Huntsekkern** [comment]
The loss in coverage is negligible (-0.5%). Adding tests for the interface would be nice, but I don't see how before you produce its actual implementation. And I don't think holding this code out of the main branch until the interface is implemented is helpful.

**Pull request merged**